atomic_reactor/plugins/exit_koji_promote.py

```diff
@@ -460,10 +460,24 @@ def add_log_type(output):
                 },
             },
         })
-        annotations = self.workflow.build_result.annotations
-        if annotations and 'digests' in annotations:
-            digests = get_digests_map_from_annotations(annotations['digests'])
-            metadata['extra']['docker']['digests'] = digests
+
+        if self.workflow.push_conf.docker_registries:
+            if self.workflow.tag_conf.unique_images:
+                image = self.workflow.tag_conf.unique_images[0]
+                tag = image.to_str(registry=False)
+                digests = {}
+
+                for registry in self.workflow.push_conf.docker_registries:
+                    if tag in registry.digests:
```

Comment:
Looking at `get_digests` in `store_metadata_in_osv3`, it seems that the keys in `registry.digests` contains the registry name.

```
        digests = {}  # repository -> digest
        for registry in self.workflow.push_conf.docker_registries:
            for image in self.workflow.tag_conf.images:
                image_str = image.to_str()
                if image_str in registry.digests:
                    digest = registry.digests[image_str]
                    digests[image.to_str(registry=False)] = digest

        return digests
```

Reply:
looking at this they are tags:

class DockerRegistry(Registry):
    """ v1/v2 docker registry """
    def __init__(self, uri, insecure=False):
        """
        :param uri: str, uri for pushing/pulling
        :param insecure: bool
        """
        super(DockerRegistry, self).__init__(uri, insecure=insecure)
        self.digests = {} # maps tags (str) to their digest, if available

Reply:
name of registry is in self.name from 'Registry'

Reply:
Yes, keys in `DockerRegistry.digests` are tags, i.e. pull specifications with no registry and without '@'; the result of Imagename.to_str(registry=False).

Reply:
Actually, what would be a great improvement would be to change that comment to make the code easier to follow:
```
-    self.digests = {} # maps tags (str) to their digest, if available
+    self.digests = {} # maps a tag (str) to a ManifestDigest instance, if available
```

Reply:
ok will change that